torch/fx/experimental/symbolic_shapes.py

```diff
@@ -4307,9 +4301,14 @@ def bound_sympy(self, expr: sympy.Expr, size_oblivious: bool = False) -> ValueRa
         var_to_range = {x: self.var_to_range.get(x, None) for x in expr.free_symbols}
         if size_oblivious:
             # Clamp values of size-like variables
+            # NB: discarding the old upper bound in intentional, per
+            # https://github.com/pytorch/pytorch/pull/123675
             for x in self.size_like & var_to_range.keys():
                 if var_to_range[x] is not None:
-                    var_to_range[x] = ValueRanges(2, sys.maxsize - 1)
+                    # NB: do NOT set upper to 2 ** 48, we're using this solely
+                    # to determine if we can do size-like replacement, the
+                    # upper bound is irrelevant here
```

Comment:
We don't do 2 ** 48 here. Should you move it to that part where we do clamp upper to 2 ** 48?

Reply:
An old version of this PR, I clamped this to 2 ** 48, but it's harmful, so I don't do it. I can try to explain it more carefully if it helps, but I don't 100% grok this part.

Reply:
you still clamp 2 ** 48 a few lines below. If you think that can be removed that'd be great as it is quite hacky.

Reply:
a few = "a hundred lines, in a completely different function" :P

The one below cannot be removed without failing tests. The problem is that u0 == sys.maxsize will fail with guard on data dependent exception, because u0 *could* be sys.maxsize and then where would that leave us. I think changing size oblivious to eliminate 0, 1 and very large integers is a good compromise.